src/client/dfs/dfs.c

```diff
@@ -2968,16 +2968,14 @@ dfs_release(dfs_obj_t *obj)
 		break;
 	default:
 		D_ERROR("Invalid entry type (not a dir, file, symlink).\n");
-		rc = EINVAL;
+		rc = -DER_IO_INVAL;
```

Comment:
this is wrong. we should return positive errno.

Reply:
oh my bad actually. we do call daos_der2errno later

Reply:
yes, exactly. I changed to always translate errors through daos_der2errno() at return.